mily/utils.py

```diff
@@ -0,0 +1,80 @@
+"""
+Various Qt Utilities
+"""
+import io
+import traceback
+
+from qtpy.QtWidgets import QWidget, QMessageBox
+
+
+def clear_layout(layout):
+    """
+    Clear a QLayout
+
+    This method acts recursively, clearing any child layouts that may be
+    contained within the layout itself. All discovered widgets are marked for
+    deletion.
+
+    Parameters
+    ----------
+    layout : QLayout
+    """
+    while layout.count():
+        child = layout.takeAt(0)
+        if child.widget():
+            child.widget().deleteLater()
+        elif child.layout():
+            clear_layout(child.layout())
+
+
+def reload_widget_stylesheet(widget, cascade=False):
+    """
+    Reload the stylesheet of a QWidget
+
+    When a ``pyqtProperty`` is updated, a widget does not automatically reapply
+    its stylesheet. This is an issue if a stylesheet is conditionally applied
+    based on the ``pyqtProperty`` value. This method handles reapplying the
+    existing stylesheet to guarantee that interface is up-to-date with any
+    changes to the contained properties
+
+    Parameters
+    ----------
+    widget: QWidget
+
+    cascade: bool, optional
+        Any child of the provided widget will also re-update its stylesheet
+    """
+    widget.style().unpolish(widget)
+    widget.style().polish(widget)
+    widget.update()
+    if cascade:
+        for child in widget.children():
+            if isinstance(child, QWidget):
```

Comment:
I'm sure there's good reason for this, but I'm curious: what children would not be `QWidget`s?

Reply:
Man, should have put a comment here... I know this `isinstance` was hard earned... I'd have to check what actually gets in that `children` call.

Reply:
You can have `QObject` or even `QLayout` being a child of a certain widget and the QObject does not offer the `.style()` stuff as this is part of `QWidget`. 
That is why this verification is required.